lib/attributesParser.js

```diff
@@ -27,15 +27,33 @@ var parser = new Parser({
 	inside: {
 		"\\s+": true, // eat up whitespace
 		">": "outside", // end of attributes
+		"style\\s*=\\s*([\"\'])": function(match, quoteType) {
+			this.styleQuoteType = quoteType;
+			return "styleAttr";
+		},
+		"style\\s*=\\s*\'": 'styleAttrSingleQuote',
 		"(([0-9a-zA-Z\\-:]+)\\s*=\\s*\")([^\"]*)\"": processMatch,
 		"(([0-9a-zA-Z\\-:]+)\\s*=\\s*\')([^\']*)\'": processMatch,
 		"(([0-9a-zA-Z\\-:]+)\\s*=\\s*)([^\\s>]+)": processMatch
+	},
+	styleAttr: {
+		"\\s+": true, // eat up whitespace
+		"\\\\[\'\"]": true, // eat escaped quotes so the end of style attribute rule doesn't find them
+		"([\"\'])": function(match, quoteType) {
+			return this.styleQuoteType === quoteType ?
+				"inside" : // end of style attribute, beginning and end quotes match
+				false;
+		},
+		"((url\\([\"\']?))([^\"\'\\)]*)[\"\']?\\)": function(match, strUntilValue, name, value, index) {
```

Comment:
This still will find mismatched beginning and ending quotes such as:
```css
url("image.png')
url('image.png")
url("image.png)
url(image.png')
```
It would be a lot more complex to handle all the cases. It should keep the extra quotes away from webpack and just leave them when the url is rewritten. User styles shouldn't have random quotes in here. I feel like it should be the user's problem if they have code like this.